Feature/KAS-2625 visie nota link

erikap opened the pull request:
Toevoegen van een computed property op agendaitem om de nota of, indien er geen nota is, de visienota te krijgen. Deze computed property wordt gebruikt in Kort Bestek zodanig dat de quick link doorverwijst naar de nota met fallback naar de visienota.

Nog enkele andere document types verhuisd van de config naar de constants files met de nodige aanpassingen in de bestaande code base.

erikap committed:
Fallback to visienota if agendaitem doesn't have a nota in kort bestek

erikap committed:
Move nota and visienota URIs to constants config

erikap committed:
Add nota and remark dossier type URIs to constants config

erikap committed:
Remove unused minute document type id from config file

erikap committed:
Move decision document type URI to constants config

erikap committed:
Move decreet document type to constants config

erikap committed:
Move BVR document type to constants config

TaneeG commented on app/utils/config.js:
```diff
@@ -50,20 +50,6 @@ export default EmberObject.create({
-  },
-  decisionDocumentTypeId: '2b73f8e2-b1f8-4cbd-927f-30c91759f08b',
-  minuteDocumentTypeId: 'e149294e-a8b8-4c11-83ac-6d4c417b079b',
   remarkId: '305E9678-8106-4C14-9BD6-60AE2032D794',
```
Je bent deze vergeten te verwijderen denk ik

erikap replied:
De `remarkId` bedoel je?  Die heb ik ook nog niet gereplaced in de code, dus kan nog niet verwijderd worden uit `utils/config.js`. 

~Er was vroeger ook nog een `DOSSIER_TYPES.REMARK` in de `utils/config.js` en die heb ik wel al  vervangen. Vandaar misschien je verwarring.~ (dat is mis, ik heb gewoon nog niet alle occurrences van `remarkId` vervangen in de code. Vandaar dat die er nog staat.

ValenberghsSven replied:
die remarkId eruit halen is hier logisch gezien buiten scope.
Sowieso als we dit willen wijzigen moeten we ook de `vl-radio-button-list` component wijzigen om `option.uri` terug te geven ipv `option.id`

erikap replied:
Dat is misschien ook de reden waarom ik hem niet meegenomen heb om niet te ver van de scope van de story af te wijken :see_no_evil: Ik laat `remarkId` dus voorlopig staan.

TaneeG replied:
Aha :D Oke

ValenberghsSven committed:
Merge branch 'development' of https://github.com/kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-2625-visie-nota-link

ValenberghsSven commented on app/models/case.js:
```diff
@@ -28,12 +28,12 @@ export default Model.extend({
   hasBvr: computed('pieces.@each', async function() {
-    const documentTypeBesluit = await this.store.findRecord('document-type', CONFIG.documentType.besluitVlaamseRegering.id);
+    const documentTypeBesluit = await this.store.findRecordByUri('document-type', CONSTANTS.DOCUMENT_TYPES.DECISION_VR);
     return await this.caseService.hasPieceOfType(this, documentTypeBesluit);
   }),
 
   hasDecreet: computed('pieces.@each', async function() {
-    const documentTypeDecreet = await this.store.findRecord('document-type', CONFIG.documentType.decreet.id);
+    const documentTypeDecreet = await this.store.findRecordByUri('document-type', CONSTANTS.DOCUMENT_TYPES.DECREET);
     return await this.caseService.hasPieceOfType(this, documentTypeDecreet);
   }),
```
deze 2 computeds zijn niet gebruikt (nog een legacy van voor gebruik van publicatie regulation-type)
Weg ermee ?

erikap replied:
OK, ik verwijder deze ongebruikte computeds nog en als Jenkins groen is merge ik de PR.

ValenberghsSven reviewed (approved):
Aangehaald op de sprintplanning: update notas feature werkt niet met die visieNota. Komt in vervolgticket.

Buiten een optionele opkuis van 2 computes is deze branch OK

erikap committed:
Remove unused computed properties from case model

erikap committed:
Refactor latestSubcase computed property on case

By using store.queryOne sorted by creation date only the relevant subcase is retrieved from the backend

ValenberghsSven committed:
KAS-2625 cleanup unused import, remove unused service

ValenberghsSven commented:
Wacht nog even jenkins af (build 4), als die groen is kunnen we mergen (build 2 had flaky test ongerelateerd aan deze change)
